### PR TITLE
CTest Updates - Fix duplicates

### PR DESCRIPTION
**Pull request by kiritigowda**
* Remove duplicate files
* Cleanup

### CODE REVIEW

**Review by LakshmiKumar23 — APPROVED** [review]

**Review by rrawther — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -164,7 +167,7 @@ if(HIP_FOUND AND Libva_FOUND)
   # make test with CTest
   enable_testing()
   include(CTest)
-  add_subdirectory(samples)
+  add_subdirectory(test)
```
Comment (AryanSalmanpour) [inline CMakeLists.txt:170]: @kiritigowda  There is a Cmake error with this change. have you tested this on your local machine?

**Review by AryanSalmanpour — CHANGES_REQUESTED** [review]

**Comment by kiritigowda** [comment]
@AryanSalmanpour -- cmake and build

```
cmake ../rocdecode/
-- The CXX compiler identification is Clang 10.0.0
-- Check for working CXX compiler: /usr/bin/clang++
-- Check for working CXX compiler: /usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- rocDecode Version -- 0.7.0
-- rocDecode Install Path -- /opt/rocm
-- rocDecode Build Type -- Release
-- Performing Test HIP_CLANG_SUPPORTS_PARALLEL_JOBS
-- Performing Test HIP_CLANG_SUPPORTS_PARALLEL_JOBS - Failed
-- Found Libva Version: 1.7.0
-- 	Libva Version Supported
-- Using Libva -- 
	Libraries:/usr/lib/x86_64-linux-gnu/libva.so 
	Includes:/usr/include
-- Using Libva-drm -- 
	Libraries:/usr/lib/x86_64-linux-gnu/libva-drm.so
-- AMD ROCm rocDecode -- CMAKE_CXX_FLAGS: -O3 -DNDEBUG -fPIC -std=gnu++17
-- AMD ROCm rocDecode -- Link Libraries: hip::host;/usr/lib/x86_64-linux-gnu/libva.so;/usr/lib/x86_64-linux-gnu/libva-drm.so
-- The C compiler identification is Clang 18.0.0
-- Check for working C compiler: /opt/rocm/llvm/bin/clang
-- Check for working C compiler: /opt/rocm/llvm/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- rocdecode-test :ROCM_PATH Set -- /opt/rocm
-- Using rocDecode -- 
	Libraries:/opt/rocm/lib/librocdecode.so 
	Includes:/opt/rocm/include/rocdecode
-- rocDecode .deb RunTime Package Requirements --  rocm-hip-runtime, libva2, libdrm-amdgpu1, mesa-amdgpu-va-drivers
-- rocDecode .deb Dev Package Requirements --  rocdecode, rocm-hip-runtime-dev, libva-dev, pkg-config, ffmpeg, libavcodec-dev, libavformat-dev, libavutil-dev
-- Configuring done
-- Generating done
-- Build files have been written to: /home/kiriti/develop/rocdecode-kiriti/build
kiriti@simon:~/develop/rocdecode-kiriti/build$ make -j8
Scanning dependencies of target rocdecode
[ 11%] Building CXX object CMakeFiles/rocdecode.dir/src/parser/roc_video_parser.cpp.o
[ 22%] Building CXX object CMakeFiles/rocdecode.dir/src/parser/hevc_parser.cpp.o
[ 44%] Building CXX object CMakeFiles/rocdecode.dir/src/parser/av1_parser.cpp.o
[ 44%] Building CXX object CMakeFiles/rocdecode.dir/src/parser/rocparser_api.cpp.o
[ 55%] Building CXX object CMakeFiles/rocdecode.dir/src/parser/avc_parser.cpp.o
[ 66%] Building CXX object CMakeFiles/rocdecode.dir/src/rocdecode/roc_decoder.cpp.o
[ 77%] Building CXX object CMakeFiles/rocdecode.dir/src/rocdecode/rocdecode_api.cpp.o
[ 88%] Building CXX object CMakeFiles/rocdecode.dir/src/rocdecode/vaapi/vaapi_videodecoder.cpp.o
[100%] Linking CXX shared library lib/librocdecode.so
[100%] Built target rocdecode
```

**Comment by kiritigowda** [comment]
CTests - @AryanSalmanpour 


CI - http://math-ci.amd.com/blue/organizations/jenkins/mainline%2Fprecheckin%2FrocDecode/detail/PR-408/2/pipeline/549
```
7/7 Test #7: video_decode-AV1 .................   Passed    5.02 sec

100% tests passed, 0 tests failed out of 7

```

File: CMakeLists.txt (hunk 1 above)
Comment (kiritigowda) [reply CMakeLists.txt:170]: @AryanSalmanpour -- see logs below. Are you seeing any issues?

Comment (AryanSalmanpour) [reply CMakeLists.txt:170]: @kiritigowda I see this error with this PR. The develop branch ToT works fine.

cd rocDecode
mkdir build;cd build

cmake ..
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:33 (project):
  The CMAKE_CXX_COMPILER:

    clang++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!

Comment (kiritigowda) [reply CMakeLists.txt:170]: Is your OS Ubuntu 22.04?

Comment (kiritigowda) [reply CMakeLists.txt:170]: @AryanSalmanpour - CI results - http://math-ci.amd.com/blue/organizations/jenkins/mainline%2Fprecheckin%2FrocDecode/detail/PR-408/2/pipeline/549/


Need to reproduce your issue as I see the AzureCI failing with the same issue

Comment (AryanSalmanpour) [reply CMakeLists.txt:170]: Yes, my OS is UB 22.04 + mainline Build#14571

Comment (kiritigowda) [reply CMakeLists.txt:170]: @AryanSalmanpour I will try on the mainline docker - [14577](http://rocm-ci.amd.com/job/compute-rocm-dkms-no-npi-hipclang/14577/)) and update you

**Comment by kiritigowda** [comment]
@AryanSalmanpour 

## Failure
```
cmake ../rocdecode/
-- The CXX compiler identification is unknown
CMake Error at CMakeLists.txt:33 (project):
  The CMAKE_CXX_COMPILER:

    clang++

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
See also "/root/build/CMakeFiles/CMakeOutput.log".
See also "/root/build/CMakeFiles/CMakeError.log".
```

## Fix

```
apt install clang
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,40 @@
 Documentation for rocDecode is available at
 [https://rocm.docs.amd.com/projects/rocDecode/en/latest/](https://rocm.docs.amd.com/projects/rocDecode/en/latest/)
 
+## rocDecode 0.7.0 (unreleased)
```
Comment (kiritigowda) [inline CHANGELOG.md]: @rrawther -- required changelog upgrade

**Comment by kiritigowda** [comment]
RHEL

```
7/7 Test #7: video_decode-AV1 .................   Passed    4.93 sec

100% tests passed, 0 tests failed out of 7

Total Test time (real) = 116.86 sec
bash-5.1# cat /etc/os-release 
NAME="Red Hat Enterprise Linux"
VERSION="9.1 (Plow)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="9.1"
PLATFORM_ID="platform:el9"
PRETTY_NAME="Red Hat Enterprise Linux 9.1 (Plow)"
ANSI_COLOR="0;31"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:redhat:enterprise_linux:9::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/red_hat_enterprise_linux/9/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 9"
REDHAT_BUGZILLA_PRODUCT_VERSION=9.1
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.1"
bash-5.1# yum info rocm-core
Updating Subscription Management repositories.
Unable to read consumer identity
Subscription Manager is operating in container mode.

This system is not registered with an entitlement server. You can use subscription-manager to register.

Last metadata expiration check: 0:24:30 ago on Wed Aug 14 20:20:31 2024.
Installed Packages
Name         : rocm-core
Version      : 6.3.0.60300
Release      : crdnnh.14577.el9
Architecture : x86_64
Size         : 32 k
Source       : rocm-core-6.3.0.60300-crdnnh.14577.el9.src.rpm
Repository   : @System
From repo    : rocm
Summary      : Radeon Open Compute (ROCm) Runtime software stack
License      : MIT
Description  : DESCRIPTION
             : ===========
             : 
             : This is an installer created using CPack (https://cmake.org). No additional installation instructions provided.
```

**Review by AryanSalmanpour — APPROVED** [review]

**Pull request merged**